rotkehlchen/exchanges/htx.py

```diff
@@ -188,24 +221,234 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        """
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while len(output := self._query(absolute_path=endpoint, options=options)) != 0:
+            result.extend(output)
+            try:
+                query_options['from'] = output[-1]['id'] + 1  # from is inclusive, so we want after
+                created_at = output[-1]['created-at']  # the results are in descending order
+            except KeyError as e:
+                log.error(f'Missing expected keys {e!s} while iterating for HTX deposits and withdrawals')  # noqa: E501
+                break
+
+            if len(output) < PAGINATION_LIMIT or created_at < start_ts:
+                break
+
+        return result
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            query_for: Literal['deposit', 'withdraw'],
+    ) -> list[AssetMovement]:
+        """
+        Process deposits/withdrawals from HTX. If any asset is unknown or we fail to
+        deserialize an entry the error is logged and we skip the entry.
+
+        This logic processes all entries returned by the API, discarding those that do not fall
+        within the range start_ts <= timestamp <= end_ts. Additionally, it does not terminate the
+        querying process until it encounters an created_at value that is lower than the start_ts.
+        """
+        log.debug(f'querying HTX deposits/withdrawals with {start_ts=}-{end_ts=}')
+        try:
+            raw_data = self._paginated_query(
+                endpoint='/v1/query/deposit-withdraw',
+                options={
+                    'type': query_for,
+                    'size': PAGINATION_LIMIT,
+                    'direct': 'next',  # sorted by descending order
+                },
+                start_ts=start_ts,
+            )
+        except RemoteError as e:
+            log.error(f'Failed to query HTX api for deposits and withdrawals due to {e!s}')
+            return []
+
+        movements = []
+        for movement in raw_data:
+            if (timestamp_raw := movement.get('created-at')) is not None:
+                timestamp = ts_ms_to_sec(TimestampMS(int(timestamp_raw)))
+                if not start_ts <= timestamp <= end_ts:
+                    continue  # skip if is not in range, could come from last page
+            else:
+                log.error(f'Could not find timestamp for HTX movement {movement}. Skipping...')
+                continue
+
+            try:
+                coin = asset_from_htx(movement['currency'])
+                movements.append(AssetMovement(
+                    timestamp=timestamp,
+                    location=Location.HTX,
+                    category=AssetMovementCategory.DEPOSIT if query_for == 'deposit' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
```

Comment:
Take that out of the for loop. `query_for` is given as an argument to the function.